src/cordova/build.action.mjs

```diff
@@ -14,95 +14,162 @@
-    ];
-
-    if (verbose) {
-      argv.push('--gradleArg=--info');
-      cordova.on('verbose', message => console.debug(`[cordova:verbose] ${message}`));
-    }
+  switch (platform + buildMode) {
+    case 'android' + 'debug':
+      return androidDebug(verbose);
+    case 'android' + 'release':
+      if (!process.env.JAVA_HOME) {
+        throw new ReferenceError('JAVA_HOME must be defined in the environment to build an Android Release!');
+      }
 
-    if (buildMode === 'release') {
       if (!(process.env.ANDROID_KEY_STORE_PASSWORD && process.env.ANDROID_KEY_STORE_CONTENTS)) {
         throw new ReferenceError(
           "Both 'ANDROID_KEY_STORE_PASSWORD' and 'ANDROID_KEY_STORE_CONTENTS' must be defined in the environment to build an Android Release!"
         );
       }
-      argv = [
-        ...argv,
-        '--keystore=keystore.p12',
+
+      return androidRelease(
+        process.env.ANDROID_KEY_STORE_PASSWORD,
+        process.env.ANDROID_KEY_STORE_CONTENTS,
+        process.env.JAVA_HOME,
+        verbose
+      );
+    case 'ios' + 'debug':
+    case 'macos' + 'debug':
+      return appleDebug(platform);
+    case 'ios' + 'release':
+    case 'macos' + 'release':
+      return appleRelease(platform);
+  }
+}
+
+async function appleDebug(platform) {
+  console.warn(`WARNING: building "${platform}" in [DEBUG] mode. Do not publish this build!!`);
+
+  return spawnStream(
+    'xcodebuild',
+    'clean',
+    '-workspace',
+    path.join(getRootDir(), 'src', 'cordova', 'apple', `${platform}.xcworkspace`),
+    '-scheme',
+    'Outline',
+    '-destination',
+    platform === 'ios' ? 'generic/platform=iOS' : 'generic/platform=macOS',
+    'build',
+    '-configuration',
+    'Debug',
+    'CODE_SIGN_IDENTITY=""',
+    'CODE_SIGNING_ALLOWED="NO"'
+  );
+}
+
+async function appleRelease(platform) {
+  return spawnStream(
+    'xcodebuild',
+    'clean',
+    '-workspace',
+    path.join(getRootDir(), 'src', 'cordova', 'apple', `${platform}.xcworkspace`),
+    '-scheme',
+    'Outline',
+    '-destination',
+    platform === 'ios' ? 'generic/platform=iOS' : 'generic/platform=macOS',
+    'archive',
+    '-configuration',
+    'Release'
+  );
+}
+
+async function androidDebug(verbose) {
+  console.warn(`WARNING: building "android" in [DEBUG] mode. Do not publish this build!!`);
+
+  return cordova.compile({
+    verbose,
+    platforms: ['android'],
+    options: {
+      argv: [
+        // Path is relative to /platforms/android/.
+        // See https://docs.gradle.org/current/userguide/composite_builds.html#command_line_composite
+        '--gradleArg=--include-build=../../src/cordova/android/OutlineAndroidLib',
+        verbose ? '--gradleArg=--info' : '--gradleArg=--quiet',
+      ],
+    },
+  });
+}
+
+const JAVA_BUNDLETOOL_VERSION = '1.8.2';
+const JAVA_BUNDLETOOL_RESOURCE_PATH = `https://github.com/google/bundletool/releases/download/1.8.2/bundletool-all-${JAVA_BUNDLETOOL_VERSION}.jar`;
```

Comment:
This makes it clearer that it's not a file in the filesystem and needs to be fetched:
```suggestion
const JAVA_BUNDLETOOL_RESOURCE_URL = `https://github.com/google/bundletool/releases/download/1.8.2/bundletool-all-${JAVA_BUNDLETOOL_VERSION}.jar`;
```